src/controllers/action-queue/display.ts

```diff
@@ -11,7 +22,17 @@ import { checkEffectIsDone } from './util/checkEffectIsDone'
 import { getEffectErrors } from './util/getEffectErrors'
 
 export async function getActionProgramDisplayInfo(account: EdgeAccount, program: ActionProgram, programState: ActionProgramState): Promise<ActionDisplayInfo> {
-  return await getActionOpDisplayInfo(account, program.actionOp, programState.effect)
+  // Assume that ActionPrograms always have a NodeActionOp at top level, for purposes of retreiving title and message as the completion strings
+  const programType = program.actionOp.type
+  if (programType !== 'seq' && programType !== 'par') throw new Error('getActionProgramDisplayInfo only supports NodeActionOps as root program type')
```

Comment:
Can't we support `LeafActionOp` types with the pure inference method?

Reply:
Ah yeah that was the point in the first place, good catch

Reply:
Actually I did try that. Right now, we can't go by inference because getActionOpDisplayInfo drills down into the leaves. When we fully isolate the display implementation we can rely more on static typing

Reply:
Actually this check isn't needed if I instead restrict the ActionProgram's root actionOp type as a NodeActionOp

Reply:
I don't see why we would ever have a `LeafActionOp` in an action program... This is just one action which doesn't require the action queue, and even if we did want a program with one action for whatever reason, adding the `NodeActionOp` could just be required overhead. I'm okay with making this change as a breakaway task.